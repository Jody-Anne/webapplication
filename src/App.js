import React, { useState } from 'react';
import axios from 'axios';
const express = require('express');
const app = express();
const mongoose = require('mongoose');

mongoose.connect('mongodb://localhost/favorites', {
  useNewUrlParser: true,
  useUnifiedTopology: true
});

const favoriteSchema = new mongoose.Schema({
  name: String,
  artist: String,
  id: Number
});

const Favorite = mongoose.model('Favorite', favoriteSchema);

app.post('/favorites', async (req, res) => {
  const favorite = new Favorite(req.body);
  try {
    await favorite.save();
    res.status(201).json(favorite);
  } catch (err) {
    console.error(err);
    res.status(500).json({ message: 'Server Error' });
  }
});

app.listen(3000, () => {
  console.log('Server is running on port 3000');
});

const express = require('express');
const app = express();
const axios = require('axios');

app.get('/search', async (req, res) => {
  const { q } = req.query;
  try {
    const result = await axios.get(`https://itunes.apple.com/search?term=${q}`);
    res.json(result.data.results);
  } catch (err) {
    console.error(err);
    res.status(500).json({ message: 'Server Error' });
  }
});

app.listen(3000, () => {
  console.log('Server is running on port 3000');
});

function Search() {
  const [query, setQuery] = useState('');
  const [results, setResults] = useState([]);

  const handleSubmit = async (e) => {
    e.preventDefault();
    const res = await axios.get(`/search?q=${query}`);
    setResults(res.data);
  }

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <input type="text" value={query} onChange={e => setQuery(e.target.value)} />
        <button type="submit">Search</button>
      </form>
      <div>
        {results.map(result => (
          <div key={result.id}>
            <h2>{result.name}</h2>
            <p>{result.artist}</p>
          </div>
        ))}
      </div>
    </div>
  );
}

export default Search;
