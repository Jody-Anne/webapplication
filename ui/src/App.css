.wrapper {
  width: 800px;
  margin: auto;
}

.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

form {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 20px;
}

input[type="text"] {
  padding: 10px;
  font-size: 18px;
  border-radius: 5px;
  border: none;
  box-shadow: 0 0 5px rgba(0, 0, 0, 0.3);
  margin-right: 10px;
}

select {
  padding: 10px;
  font-size: 18px;
  border-radius: 5px;
  border: none;
  box-shadow: 0 0 5px rgba(0, 0, 0, 0.3);
  margin-right: 10px;
}

button[type="submit"] {
  padding: 10px;
  font-size: 18px;
  border-radius: 5px;
  border: none;
  background-color: lavenderblush;
  color: black;
  cursor: pointer;
  transition: background-color 0.3s ease;
}

button[type="submit"]:hover {
  background-color: lavenderblush;
}

/* Styles for the search results */
ul {
  list-style: none;
  margin: 0;
  padding: 0;
}

li {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px;
  margin-bottom: 10px;
  background-color: #f7f7f7;
  border-radius: 5px;
  box-shadow: 0 0 5px rgba(0, 0, 0, 0.3);
}

li button {
  padding: 10px;
  font-size: 16px;
  border-radius: 5px;
  border: none;
  background-color:lavenderblush;
  color: black;
  cursor: pointer;
  transition: background-color 0.3s ease;
}

li button:hover {
  background-color: #794a6a;
}
