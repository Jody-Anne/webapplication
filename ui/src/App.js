import './App.css';
import ItunesFavourites from './ItunesFavourites';

function App() {
  return (
    <div className='Header'>
      <div className='wrapper'>
        <header className='Head'>
          <h1>iTunes Search</h1>
        </header>
        <ItunesFavourites/>
      </div>
    </div>
  );
}

export default App;
