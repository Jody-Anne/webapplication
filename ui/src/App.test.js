import renderer from 'react-test-renderer';
import App from './App';

test('create a snapshot and test that it matches', () => {
  const app = renderer.create(<App />).toJSON();
  expect(app).toMatchSnapshot();
});
