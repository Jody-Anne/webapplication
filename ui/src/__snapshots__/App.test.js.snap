// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a snapshot and test that it matches 1`] = `
<div
  className="Header"
>
  <div
    className="wrapper"
  >
    <header
      className="Head"
    >
      <h1>
        iTunes Search
      </h1>
    </header>
    <div>
      <div>
        <h3>
          Search for books or music on the iTunes store 😃:
        </h3>
        <form
          onSubmit={[Function]}
        >
          <input
            name="query"
            type="text"
          />
          <select
            name="media"
            onChange={[Function]}
            value="all"
          >
            <option
              value="all"
            >
              All
            </option>
            <option
              value="music"
            >
              🎶 Music
            </option>
            <option
              value="audiobook"
            >
              🕮 Audiobooks
            </option>
            <option
              value="ebook"
            >
              📚 Books
            </option>
            <option
              value="tvShow"
            >
              📺 TV Shows
            </option>
            <option
              value="musicVideo"
            >
              📹 Music Videos
            </option>
            <option
              value="podcast"
            >
              🎙️ Podcasts
            </option>
            <option
              value="movie"
            >
              🎞️ Movie
            </option>
          </select>
          <button
            type="submit"
          >
            Search
          </button>
        </form>
        <div
          className="results-container"
        >
          <ul
            className="search-results"
          />
        </div>
      </div>
      <div
        className="favourites-container"
      >
        <h3>
          Favourites🫶:
        </h3>
        <ul
          className="favourites-list"
        >
          <p>
            No favourites yet!
          </p>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
